landing_page/app/controllers/microposts_controller.rb

```diff
@@ -38,7 +38,12 @@ def recover
     helper_method :recover
 
     def index 
-        @microposts = Micropost.kept.order(created_at: :desc) 
+        
+        if params[:archived] == "true"
+            @microposts = Micropost.order(created_at: :desc).kept.where("created_at <= ?", Time.now-7.days)
+        else
+            @microposts = Micropost.kept.where("created_at > ?", Time.now-7.days).order(created_at: :desc)
```

Comment:
Inverting the order of the operations does not solve the readability issue, actually it makes the code even harder to read, because the reader will get more confused until realising that they are identical, with the exception of the seven-day test.